frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java

```diff
@@ -140,9 +139,15 @@ private void suggestBranchDeletion(AnActionEvent anActionEvent, String branchNam
 
     new Task.Backgroundable(project, "Deleting branch if required...") {
       @Override
+      @UIThreadUnsafe
       public void run(ProgressIndicator indicator) {
         if (slideOutOptions != null) {
           handleBranchDeletionDecision(project, branchName, gitRepository, anActionEvent, slideOutOptions.shouldDelete());
+
+          if (slideOutOptions.shouldRemember()) {
```

Comment:
@mkondratek since you're the original author... if you could eyeball this area if there won't be any unintended consequences of this change (race conditions?)

Reply:
lgtm

Reply:
Keeping this piece of code where it was would cause a test failure, right?

Reply:
Yes, exactly: ArchUnit would complain about calling `@UIThreadUnsafe` method from `@UIEffect` (or actually, just from non-`@UIThreadUnsafe`) method